03_letters/willshen8/letters_test.go

```diff
@@ -0,0 +1,44 @@
+package main
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestMainOutput(t *testing.T) {
+	var buf bytes.Buffer
+	out = &buf
+	main()
+	expected := "a:2\nb:1\n"
+	actual := buf.String()
+
+	if expected != actual {
+		t.Errorf("Unexpected output in main(). Expected = %v Actual = %v", expected, actual)
+	}
+}
+
+func TestLetterFrequency(t *testing.T) {
```

Comment:
Great work! 
I guess it's personal preference, bit I prefer table test to be outside testing functions, that way they can be used across multiple testing functions. 
in this case it doesn't really make a difference, although it's nice to keep in mind

Reply:
Ok, thanks. I'll keep this in mind for future testing.